packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -136,11 +136,11 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
   } else if (import.meta.server) {
     const nuxtApp = useNuxtApp()
     const writeFinalCookieValue = () => {
-      if (opts.readonly || isEqual(cookie.value, cookies[name])) { return }
+      if (opts.readonly || hash(cookie.value) === hash(cookies[name])) { return }
       nuxtApp._cookies ||= {}
       if (name in nuxtApp._cookies) {
         // do not append a second `set-cookie` header
-        if (isEqual(cookie.value, nuxtApp._cookies[name])) { return }
+        if (hash(cookie.value) === hash(nuxtApp._cookies[name])) { return }
```

Comment:
You can use isEqual util here. It is faster since does only === or serialize (no sha256 is needed)